[Bug]: --engine-use-ray is broken.

### Your current environment

N/A (happens in all env)

### 🐛 Describe the bug

```
python -m vllm.entrypoints.openai.api_server --model "facebook/opt-125m" --dtype auto --api-key token-abc123 --engine-use-ray
```

When I start an openai server with engine-use-ray, I found it crashes with the following error

```
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/ray/anaconda3/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/ray/default/vllm/vllm/entrypoints/openai/api_server.py", line 157, in <module>
    engine = AsyncLLMEngine.from_engine_args(
  File "/home/ray/default/vllm/vllm/engine/async_llm_engine.py", line 347, in from_engine_args
    engine = cls(
  File "/home/ray/default/vllm/vllm/engine/async_llm_engine.py", line 311, in __init__
    self.engine = self._init_engine(*args, **kwargs)
  File "/home/ray/default/vllm/vllm/engine/async_llm_engine.py", line 413, in _init_engine
    cache_config = args[1]
IndexError: tuple index out of range
```

I suspect an "off-by-one" index error in the worker.

I am seeing what looks like a related failure when passing "--tensor-parallel-size 4"

I added some instrumentation to the failure point:

vllm/worker/worker.py
```
...
309 def _check_if_gpu_supports_dtype(torch_dtype: torch.dtype):                                                             
310     # Check if the GPU supports the dtype.                                                                              
311     if torch_dtype == torch.bfloat16:                                                                                   
312         print("Current Device:")                                                                                        
313         print(torch.cuda.current_device(), "of", torch.cuda.device_count())                                             
314         compute_capability = torch.cuda.get_device_capability()
...
```

```
(RayWorkerVllm pid=2098) INFO 04-16 01:10:45 selector.py:28] Using FlashAttention backend.
Current Device:
0 of 4
(RayWorkerVllm pid=2098) Current Device:
(RayWorkerVllm pid=2098) 1 of 1
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50] Error executing method init_device. This might cause deadlock in distributed execution.
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50] Traceback (most recent call last):
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50]   File "/home/dinalt/ai_assets/code/vllm/vllm/engine/ray_utils.py", line 43, in execute_method
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50]     return executor(*args, **kwargs)
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50]   File "/home/dinalt/ai_assets/code/vllm/vllm/worker/worker.py", line 99, in init_device
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50]     _check_if_gpu_supports_dtype(self.model_config.dtype)
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50]   File "/home/dinalt/ai_assets/code/vllm/vllm/worker/worker.py", line 314, in _check_if_gpu_supports_dtype
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50]     compute_capability = torch.cuda.get_device_capability()
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50]   File "/home/dinalt/.local/lib/python3.10/site-packages/torch/cuda/__init__.py", line 439, in get_device_capability
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50]     prop = get_device_properties(device)
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50]   File "/home/dinalt/.local/lib/python3.10/site-packages/torch/cuda/__init__.py", line 456, in get_device_properties
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50]     raise AssertionError("Invalid device id")
(RayWorkerVllm pid=2098) ERROR 04-16 01:10:45 ray_utils.py:50] AssertionError: Invalid device id
```

You can see that the device index assigned to the worker is "1 of 1," where the only valid index is '0' for a single assigned GPU.

I'll see if I can root-cause it...

--- a/tests/async_engine/test_api_server.py
+++ b/tests/async_engine/test_api_server.py
@@ -46,7 +46,7 @@ def api_server(tokenizer_pool_size: int, engine_use_ray: bool,
 
 @pytest.mark.parametrize("tokenizer_pool_size", [0, 2])
 @pytest.mark.parametrize("worker_use_ray", [False, True])
-@pytest.mark.parametrize("engine_use_ray", [True])
+@pytest.mark.parametrize("engine_use_ray", [False, True])
 def test_api_server(api_server, tokenizer_pool_size: int, worker_use_ray: bool,
                     engine_use_ray: bool):
     """

--- a/tests/async_engine/test_api_server.py
+++ b/tests/async_engine/test_api_server.py
@@ -25,21 +25,30 @@ def _query_server_long(prompt: str) -> dict:
 
 
 @pytest.fixture
-def api_server(tokenizer_pool_size: int):
+def api_server(tokenizer_pool_size: int, engine_use_ray: bool,
+               worker_use_ray: bool):
     script_path = Path(__file__).parent.joinpath(
         "api_server_async_engine.py").absolute()
-    uvicorn_process = subprocess.Popen([
+    commands = [
         sys.executable, "-u",
         str(script_path), "--model", "facebook/opt-125m", "--host",
         "127.0.0.1", "--tokenizer-pool-size",
         str(tokenizer_pool_size)
-    ])
+    ]
+    if engine_use_ray:
+        commands.append("--engine-use-ray")
+    if worker_use_ray:
+        commands.append("--worker-use-ray")
+    uvicorn_process = subprocess.Popen(commands)
     yield
     uvicorn_process.terminate()
 
 
 @pytest.mark.parametrize("tokenizer_pool_size", [0, 2])
-def test_api_server(api_server, tokenizer_pool_size: int):
+@pytest.mark.parametrize("worker_use_ray", [False, True])
+@pytest.mark.parametrize("engine_use_ray", [True])
+def test_api_server(api_server, tokenizer_pool_size: int, worker_use_ray: bool,
+                    engine_use_ray: bool):
     """
     Run the API server and test it.
 

--- a/vllm/engine/async_llm_engine.py
+++ b/vllm/engine/async_llm_engine.py
@@ -333,8 +333,7 @@ class AsyncLLMEngine:
         if engine_config.device_config.device_type == "neuron":
             raise NotImplementedError("Neuron is not supported for "
                                       "async engine yet.")
-        elif (engine_config.parallel_config.worker_use_ray
-              or engine_args.engine_use_ray):
+        elif engine_config.parallel_config.worker_use_ray:
             initialize_ray_cluster(engine_config.parallel_config)
             from vllm.executor.ray_gpu_executor import RayGPUExecutorAsync
             executor_class = RayGPUExecutorAsync
@@ -410,8 +409,8 @@ class AsyncLLMEngine:
         else:
             # FIXME(woosuk): This is a bit hacky. Be careful when changing the
             # order of the arguments.
-            cache_config = args[1]
-            parallel_config = args[2]
+            cache_config = kwargs["cache_config"]
+            parallel_config = kwargs["parallel_config"]
             if parallel_config.tensor_parallel_size == 1:
                 num_gpus = cache_config.gpu_memory_utilization
             else:

